test/clj/rems/test_browser.clj

```diff
@@ -489,15 +508,120 @@
+      (select-option "Workflow" (btu/context-get :workflow-name))
+      (select-option "Resource" (btu/context-get :resid))
+      (select-option "Form" (btu/context-get :form-name))
+      (btu/screenshot (io/file btu/reporting-dir "about-to-create-catalogue-item.png"))
+      (btu/scroll-and-click :save)
+      (btu/wait-visible {:tag :h1 :fn/text "Catalogue item"})
+      (btu/wait-page-loaded)
+      (btu/screenshot (io/file btu/reporting-dir "created-catalogue-item.png"))
+      (is (str/includes? (btu/get-element-text {:css ".alert-success"}) "Success"))
+      (is (= {"Organization" "NBN"
+              "Title (EN)" (str (btu/context-get :catalogue-item-name) " EN")
+              "Title (FI)" (str (btu/context-get :catalogue-item-name) " FI")
+              "Title (SV)" (str (btu/context-get :catalogue-item-name) " SV")
+              "More info (EN)" ""
+              "More info (FI)" ""
+              "More info (SV)" ""
+              "Workflow" (btu/context-get :workflow-name)
+              "Resource" (btu/context-get :resid)
+              "Form" (btu/context-get :form-name)
+              "Active" ""
+              "End" ""}
+             (dissoc (slurp-fields :catalogue-item)
+                     "Start")))
+      (btu/scroll-and-click {:fn/text "Enable"})
+      (btu/wait-page-loaded)
+      (is (str/includes? (btu/get-element-text {:css ".alert-success"}) "Success"))
+      (go-to-admin-catalogue)
+      (is (some #(= {"workflow" (btu/context-get :workflow-name)
+                     "resource" (btu/context-get :resid)
+                     "form" (btu/context-get :form-name)
+                     "name" (str (btu/context-get :catalogue-item-name) " EN")}
+                    (select-keys % ["resource" "workflow" "form" "name"]))
+                (slurp-rows :catalogue))))))
 
 (deftest test-create-catalogue-item
   (btu/with-postmortem {:dir btu/reporting-dir}
-    (login-as "owner")
-    (btu/context-assoc! :license-name (str "Browser Test License " (btu/get-seed))
-                        :resid (str "browser.testing.resource/" (btu/get-seed)))
-    (create-license)
-    (create-resource)
-    (testing "create form") ; TODO
-    (testing "create workflow") ; TODO
-    (testing "create catalogue item"))) ; TODO
+    (testing "create objects"
+      (login-as "owner")
+      (btu/context-assoc! :license-name (str "Browser Test License " (btu/get-seed))
+                          :resid (str "browser.testing.resource/" (btu/get-seed))
+                          :form-name (str "Browser Test Form " (btu/get-seed))
+                          :workflow-name (str "Browser Test Workflow " (btu/get-seed))
+                          :catalogue-item-name (str "Browser Test Catalogue Item " (btu/get-seed)))
+      (create-license)
+      (create-resource)
+      (create-form)
+      (create-workflow)
+      (create-catalogue-item))
+    (testing "check that catalogue item is visible for applicants"
```

Comment:
Enabling could happen here. We certainly don't use it elsewhere, but we could check that the catalogue item is visible only after enabling. That sounds valuable enough to test because looks like it's easy!

Reply:
ok